### PR TITLE
refactor(layout): update readme

### PR DESCRIPTION
**Pull request by rinaok**

### CODE REVIEW

**Comment by rinaok** [comment]
@rachelbt @yinonov I'm trying to document layout in a more understandable way. How can I show the margins is gutters?

**Comment by rachelbt** [comment]
> @rachelbt @yinonov I'm trying to document layout in a more understandable way. How can I show the margins is gutters?

It is so important the layout docs! 🦾 

I think that the easy way to show the margin is with image, and one code example. 
I also thing that we can give each layout a wrapper to style it and we can insert the CSS inside the iframe css to avoid code duplication, in all of our code preview.

image

```
<div class="box-wrapper">
<vwc-layout>
  <div class="box">default</div>
  <div class="box">default</div>
</vwc-layout>
</div>
```
in `iframe.css`:
```
.box-wrapper{
  background-image: linear-gradient(to right, var(--vvd-color-info-30), var(--vvd-color-cta-30));
  padding: 1rem;
  border-radius: 6px;
}
.box-wrapper + .box-wrapper { margin-top: 2rem;}
.box {    
  border-radius: 6px;
  background-color: var(--vvd-color-canvas)
  color: var(--vvd-color-on-canvas);
  padding: 1rem;
  text-align: center;
}
```

**Comment by rinaok** [comment]
> > @rachelbt @yinonov I'm trying to document layout in a more understandable way. How can I show the margins is gutters?
> 
> It is so important the layout docs! 🦾
> 
> I think that the easy way to show the margin is with image, and one code example. I also thing that we can give each layout a wrapper to style it and we can insert the CSS inside the iframe css to avoid code duplication, in all of our code preview.
> 
> image
> 
> ```
> <div class="box-wrapper">
> <vwc-layout>
>   <div class="box">default</div>
>   <div class="box">default</div>
> </vwc-layout>
> </div>
> ```
> 
> in `iframe.css`:
> 
> ```
> .box-wrapper{
>   background-image: linear-gradient(to right, var(--vvd-color-info-30), var(--vvd-color-cta-30));
>   padding: 1rem;
>   border-radius: 6px;
> }
> .box-wrapper + .box-wrapper { margin-top: 2rem;}
> .box {    
>   border-radius: 6px;
>   background-color: var(--vvd-color-canvas)
>   color: var(--vvd-color-on-canvas);
>   padding: 1rem;
>   text-align: center;
> }
> ```

Wow! Looks awesome 👏 😎 🤩

File: libs/components/src/lib/layout/README.md [hunk 1]
```diff
@@ -21,134 +21,132 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
 
 ## Auto-sizing
 
-Use the `auto-sizing` attribute to set the initial preferred auto-sizing.
+Use the `auto-sizing` attribute to set the initial preferred `auto-sizing`.
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:24]: shed more light?
[Auto-Sizing Columns in CSS Grid: `auto-fill` vs `auto-fit` - CSS ...https://css-tricks.com › auto-sizing-columns-css-grid-aut...](https://css-tricks.com/auto-sizing-columns-css-grid-auto-fill-vs-auto-fit/)

or 

https://stackoverflow.com/a/46227025/3027703

**Review by yinonov — CHANGES_REQUESTED** [review]

**Review by yinonov — COMMENTED** [review]
please see https://github.com/Vonage/vivid-3/pull/300#pullrequestreview-936037716

This is a great initiative and improves the doc a lot
but in practice it adds more code to read within the code snippets and can be avoided by applying this elsewhere in our implementation details

**Comment by rinaok** [comment]
> please see [#300 (review)](https://github.com/Vonage/vivid-3/pull/300#pullrequestreview-936037716)
> 
> This is a great initiative and improves the doc a lot but in practice it adds more code to read within the code snippets and can be avoided by applying this elsewhere in our implementation details

@yinonov Is there a way to do it without adding <div class="box-wrapper"> and <div class="box">?

**Review by YonatanKra — CHANGES_REQUESTED** [review]
First - the first example should be explained - what is the default? How does it work?
Second - I think `auto-sizing` is too-much-info.  let's move it to the end.
Third - once the defaults are explained, I think it is easier to explain the variations.

File: libs/components/src/lib/layout/README.md [hunk 2]
```diff
@@ -23,6 +23,12 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
 
 Use the `auto-sizing` attribute to set the initial preferred `auto-sizing`.
 
+When the [repeat()](#css-custom-properties) function is set to `fit` or `fill`, the grid container creates as many grid tracks (columns/rows) as possible without overflowing the container.
+
+With `fit`, when there are not enough grid items to fill the number of tracks created, those empty tracks are collapsed.
+
+With `fill`, everything is the same as `fit`, except empty tracks are not collapsed.
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:30]: well written! bravo

File: libs/components/src/lib/layout/README.md [hunk 3]
```diff
@@ -149,4 +155,31 @@ Use the `column-basis` attribute to set the initial preferred `measure` of a col
 
 | Name                           | Description                                      | Usage                                              |
 | ------------------------------ | ------------------------------------------------ | -------------------------------------------------- |
-| --layout-grid-template-columns | controls the grid-template-columns of the layout | repeat(`auto-sizing`, minmax(`column-basis`, 1fr)) |
+| `--layout-grid-template-columns` | controls the grid-template-columns of the layout | repeat(`auto-sizing`, minmax(`column-basis`, 1fr)) |
+
+```html preview
+<div class="box-wrapper">
+  <vwc-layout style="--layout-grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));">
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:162]: why overriding the style if we only want to show fill / fit difference?

should preserve dedicated doc section for this css variable member

Comment (yinonov) [reply libs/components/src/lib/layout/README.md:162]: never mind, I now see they don't relate

**Review by YonatanKra — APPROVED** [review]

File: libs/components/src/lib/layout/README.md [hunk 4]
```diff
@@ -19,136 +19,171 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
...
+### Grid-template-columns
+
+Controls the grid-template-columns of the layout.
 
-| Name                         | Description                                      | Usage                                              |
-| ---------------------------- | ------------------------------------------------ | -------------------------------------------------- |
-| --layout-grid-template-columns | controls the grid-template-columns of the layout | repeat(`auto-sizing`, minmax(`column-basis`, 1fr)) |
+- `--layout-grid-template-columns`
+- Type: repeat(`auto-sizing`, minmax(`column-basis`, 1fr))
+- Default: auto
```
Comment (rinaok) [inline libs/components/src/lib/layout/README.md:162]: @yinonov what should be the default?

File: libs/components/src/lib/layout/README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# vwc-layout
+# layout
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:1]: maybe the first example of images should also be replaced with card. the images aren't fluid and don't fully stretch

**Review by yinonov — CHANGES_REQUESTED** [review]

**Comment by rachelbt** [comment]
Looks really good!
The only thing I would change is the preview of the column spacing - the demo div wrapper padding is too big.
Maybe you can add a variation for the demo-gradient class of tight like this:
.demo-gradient {
/*gradient style*/
&.tight {padding: 0.2rem}
}
to get this:
image

File: change/@vonage-vivid-fc53a33b-c30e-4be2-823f-e64619c0e655.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (yinonov) [inline change/@vonage-vivid-fc53a33b-c30e-4be2-823f-e64619c0e655.json:1]: is this file needed?

File: libs/components/src/lib/layout/README.md [hunk 7]
```diff
@@ -10,145 +10,192 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
...
 ```
 
 ## Column-basis
 
-Use the `column-basis` attribute to set the initial preferred measure of a column.
+Use the `column-basis` attribute to set the initial preferred `measure` of a column. 
+Use `block` to stack elements one above the other.
 
-- Type: `'base-small'` | `'base'` | `'base-large'`
+- Type: `'base-small'` | `'base'` | `'base-large'` | `'block'`
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:101]: this will change very soon

**Review by yinonov — COMMENTED** [review]
the ui tests seem poor and doesn't reflect enough scenarios

File: libs/components/src/lib/layout/README.md [hunk 8]
```diff
@@ -10,145 +10,192 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
...
+<div class="demo-gradient">
+  <vwc-layout column-spacing="large">
+    <vwc-card elevation="2" text="large"></vwc-card>
+    <vwc-card elevation="2" text="large"></vwc-card>
+  </vwc-layout>
+</div>
 ```
 
-## Column Basis
+## Column-basis
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:96]:
```suggestion
## Column Basis
```

File: libs/components/src/lib/layout/README.md [hunk 9]
```diff
@@ -10,145 +10,192 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
...
+    <vwc-card elevation="2" text="block"></vwc-card>
+    <vwc-card elevation="2" text="block"></vwc-card>
+    <vwc-card elevation="2" text="block"></vwc-card>
+    <vwc-card elevation="2" text="block"></vwc-card>
+  </vwc-layout>
+</div>
 ```
 
-### Medium
+## Auto-sizing
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:139]:
```suggestion
## Auto Sizing
```

File: libs/components/src/lib/layout/README.md [hunk 10]
```diff
@@ -10,145 +10,192 @@ Using layout, elements can be arranged easily in a variety of sizes and spaces t
...
-  <div style="border: 1px solid">base-large</div>
-  <div style="border: 1px solid">base-large</div>
+    <vwc-card elevation="2" text="large"></vwc-card>
+    <vwc-card elevation="2" text="large"></vwc-card>
 </vwc-layout>
+</div>
 ```
 
-## Column Spacing
+## Column-spacing
```
Comment (yinonov) [inline libs/components/src/lib/layout/README.md:68]:
```suggestion
## Column Spacing
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: change/@vonage-vivid-fc53a33b-c30e-4be2-823f-e64619c0e655.json (hunk 6 above)
Comment (rinaok) [reply change/@vonage-vivid-fc53a33b-c30e-4be2-823f-e64619c0e655.json]: removed

**Comment by rinaok** [comment]
> 

The ui tests are without the stying

**Comment by rinaok** [comment]
> the ui tests seem poor and doesn't reflect enough scenarios

@yinonov The ui tests are without the styling but they are the same as in the docs

**Review by yinonov — APPROVED** [review]

**Pull request merged**